packages/server/src/provider/in-memory-provider/flag-configuration.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * Don't export types from this file publicly.
+ * It might cause confusion since these types are not a part of the general API,
+ * but just for the in-memory provider.
+ */
+import { EvaluationContext, JsonValue } from '@openfeature/shared';
+
+type Variants<T> = Record<string, T>;
+
+export type Flag = {
```

Comment:
Could this live in shared or will it be slightly different on the client?

Reply:
I think this will live in shared when we implement it in web. We can do some commonization then

Reply:
From what I understood from the web version there is no need for the `EvaluationContext`. We could either have a base `Flag` type in shared and it being internally extended as needed or have one with all properties and use `Omit` to specialize.

Reply:
@luizgribeiro there's still `EvaluationContext` in web, its just not passed as a parameter from the flag evaluation API that users use. Instead it's set globally, representing a logged-in user, typically. The backend system and provider still receives a context though.